### ISSUE TITLE
`wasm-smith`: add method to export everything

### ISSUE THREAD

**Issue by fitzgen** [bytecodealliance/wasm-tools#654]
Add a `export_everything` method to `Module` that does a post-generation pass to export everything in the module.

Various Wasmtime fuzz targets will call functions *if they are exported* or hash memories *if they are exported*.

An `export_everything` method would give us an easy way to export everything.

**Comment by alexcrichton** [comment]
I don't think so because the same item can be reexported multiple times, meaning that an item still may not be picked to get reexported.
